### ISSUE TITLE
Camera use causes crash and reboot

### ISSUE THREAD

**Issue by CoffeeMan1ac** [GrapheneOS/os-issue-tracker#4323]
The system crashes and the phone reboots randomly when camera is opened. When trying to take pictures/record videos (any camera app; native or gcam), opening qr code scanner app (uses camera), recording video in Telegram, starting video calls through Telegram, trying to rent a bike by scanning qr code within bike renting app etc. The screen just goes black for a while, then the phone reboots.

I have made three bug reports right after the phone rebooted from these crashes and able to attach them if needed. I am also able to make it crash and reboot and then share the logs.

Unable to find any reason why this started happening. Did not mess with camera settings. The last thing involving camera use right before the crashes started happening was id verificarion through Persona in Firefox browser.

I have tinkered a bit, but it did not help. I tried to install Pixel Camera Services from Google Play (it shows already installed in the settings, but does not show installed on Google Play), but it returned an error. I uninstalled gcam (and could not install it back from Google Play - returns an error), but crash-reboots still persist.

**Comment by CoffeeMan1ac** [comment]
> > I have tinkered a bit, but it did not help. I tried to install Pixel Camera Services from Google Play (it shows already installed in the settings, but does not show installed on Google Play), but it returned an error. I uninstalled gcam (and could not install it back from Google Play - returns an error), but crash-reboots still persist.
> 
> Tinkering with things is a likely source of an issue since it works by default and this isn't happening for others.

I ment, I tried to solve the crash-reboots by trying to install Pixel Camera Services from Google Play, but it could not install.
I then uninstalled gcam, and tried to reinstall it from Google Play, but it returns an error just like that for Pixel Camera Services

**Comment by CoffeeMan1ac** [comment]
> Pixel Camera Services is already included in the OS and doesn't need to be installed. Installing Pixel Camera will work fine. Perhaps you have a modded Pixel Camera app installed in other profiles or you have a newer version than what the Play Store is giving you in another profile.

I dont have multiple users enabled. I have a work profile, but only ever installed Aurora Store, Instagram and WhatsApp there, so nothing like Camera Services.
My gcam was installed from google play, as well, I have never installed anything outside Google Play, Aurora or F-Droid stores.
The gcam is not the main issue. The crash-reboots persist even without gcam installed.

**Comment by CoffeeMan1ac** [comment]
@jmvermeulen update. The camera is still broken, whenever I use it in any app, it may work, but the odds are that it's going to crash, and reboot after ~30 seconds.
Until today, I haven't gotten any crash messages (I guess the system doesn't have time to notify of crash as the phone is immediately shut off). Today, I don't know why or how, I haven't used the camera (too paranoid), I did get a crash notification and it looks similar to ones in #3840 and #3761 and #3823. Here:
`
type: crash
osVersion: google/akita/akita:14/AP2A.240705.005/2024071200:user/release-keys
uid: 1000 (u:r:hal_camera_default:s0)
cmdline: /apex/com.google.pixel.camera.hal/bin/hw/android.hardware.camera.provider@2.7-service-google
processUptime: 0s

abortMessage: Check failed: it != this->end() 

signal: 6 (SIGABRT), code -1 (SI_QUEUE)
threadName: RunnerN:P+:A:0
MTE: not enabled

backtrace:
    /apex/com.android.runtime/lib64/bionic/libc.so (abort+168, pc 66ac8)
    /system/lib64/liblog.so (__android_log_default_aborter+16, pc 6360)
    /apex/com.google.pixel.camera.hal/lib64/libbase.so (android::base::LogMessage::~LogMessage()+352, pc 171a0)
    /apex/com.google.pixel.camera.hal/lib64/liblyric_hwl.so (pc b4b9d0)
    /apex/com.google.pixel.camera.hal/lib64/liblyric_hwl.so (pc b49354)
    /apex/com.google.pixel.camera.hal/lib64/liblyric_hwl.so (pc b48d64)
    /apex/com.google.pixel.camera.hal/lib64/liblyric_hwl.so (pc b466b4)
    /apex/com.google.pixel.camera.hal/lib64/liblyric_hwl.so (pc b3a0b4)
    /apex/com.google.pixel.camera.hal/lib64/liblyric_hwl.so (pc b3a348)
    /apex/com.google.pixel.camera.hal/lib64/liblyric_hwl.so (pc e529b8)
    /apex/com.google.pixel.camera.hal/lib64/liblyric_hwl.so (pc e4f184)
    /apex/com.google.pixel.camera.hal/lib64/liblyric_hwl.so (pc f002b0)
    /apex/com.google.pixel.camera.hal/lib64/liblyric_hwl.so (pc f34e54)
    /apex/com.google.pixel.camera.hal/lib64/liblyric_hwl.so (pc f368d8)
    /apex/com.google.pixel.camera.hal/lib64/liblyric_hwl.so (pc edb510)
    /apex/com.google.pixel.camera.hal/lib64/liblyric_hwl.so (pc a397a0)
    /apex/com.google.pixel.camera.hal/lib64/liblyric_hwl.so (pc 59a7a8)
    /apex/com.google.pixel.camera.hal/lib64/liblyric_hwl.so (pc 59a6e8)
    /apex/com.android.runtime/lib64/bionic/libc.so (__pthread_start(void*)+204, pc 795dc)
    /apex/com.android.runtime/lib64/bionic/libc.so (__start_thread+68, pc 69fa4)
`

Does it mean the phone has to be repaired?

**Comment by Aquathing** [comment]
Yes, my phone is currently in the RMA process for this exact error. Google couldn't diagnose it and emailed me asking me what the problem was, I replied with the crash log and how to reproduce, and I'm waiting for them to get back to me.

**Comment by Aquathing** [comment]
Google just cancelled the repair order, stating that the phone has no issues.
![image](https://github.com/user-attachments/assets/504742fb-2794-439a-8526-ae3d4d78a435)

**Comment by Aquathing** [comment]
@thestinger while they analyze my report, could this be caused by one of the following?
- Camera app downloaded from Aurora Store (even though I recall this happening on Telegram too, which shouldn't use the Camera app directly)
- Some debug settings I do not recall having enabled?
- Some memory bug which could be causing the ASSERT to fail? It'd be nice to know what "it" and "this" are, but being closed source we can only guess, right?

**Comment by Aquathing** [comment]
> You probably just said something that caused them to wrongly believe it's only a software issue that will go away.

I have told them that the issue happens daily and provided them with the crash logs. 

My guess is that they couldn't reproduce it for some reason. 
If you remember my other issue, this stopped happening for two days after updating to Android 15 and I could reproduce it only after opening the camera hundreds of times, after which it became daily again. Perhaps reinstalling the Stock OS is causing the same issue, and they do not have the time to reproduce it like I did. Sounds strange, but I'm really out of guesses.

I'll keep this ticket updated with what Google says in hope to help others too, if this is not an issue for you.

**Comment by Aquathing** [comment]
I understand, I have suggested the person at Google who first contacted me asking for details that there could be some loose connection which causes the issues only in specific positions or movements. I will keep you updated.

Thank you for your fast answer, I will donate to the project as soon as it becomes financially viable for me!

**Comment by Aquathing** [comment]
Got back from Google.

Escalation team says this was done because they could not find an issue with the phone, which we already knew.

I replied to them asking to please have a more in-depth check, specifying that this started happening after I got the phone back from Google themselves from my first RMA!

**Comment by Aquathing** [comment]
This is what I told them.
![image](https://github.com/user-attachments/assets/db88e6bd-fa91-4f9a-819e-9e27e2e3fa0f)

(I just noticed the extra "not" in the last phrase of the third paragraph, but they'll probably understand what I'm saying anyway or not even notice it at all.)

**Comment by Aquathing** [comment]
I just received my phone, I'll take this as a chance to see if this happens with StockOS too.

**Comment by Aquathing** [comment]
**Update:** As soon as I got the phone, I tried to make the camera crash on Stock OS for 2 and a half days with a total of more than 300 attempts.

I installed GrapeheneOS, removed Aurora Store and the GCAM version installed with it. Set up play store and installed gcam from there, so far, it has only crashed 3 times that I can remember of and without producing any crash logs, there was no black screen, I could actually see the camera feed, so this looks like a different issue.
